sklearn/utils/_repr_html/params.py

```diff
@@ -2,11 +2,64 @@
 # SPDX-License-Identifier: BSD-3-Clause
 
 import html
+import re
 import reprlib
 from collections import UserDict
+from urllib.parse import quote
 
 from sklearn.utils._repr_html.base import ReprHTMLMixin
 
+CLASS_DOC_URL_PREFIX = "https://scikit-learn.org/{doc_version}/modules/generated/"
+
+
+def link_to_param_doc(estimator_type, param_name):
+    """URL to the relevant section of the docstring using a Text Fragment
+
+    https://developer.mozilla.org/en-US/docs/Web/URI/Reference/Fragment/Text_fragments
+    """
+
+    import sklearn
+
+    if hasattr(estimator_type, "__module__"):
+        module_name = estimator_type.__module__
+    else:
+        module_name = None
+
+    if module_name is None or not module_name.startswith("sklearn."):
+        # Not a scikit-learn estimator. Do not link to the scikit-learn
+        # documentation.
+        return None
+
+    if ".dev" in sklearn.__version__:
+        doc_version = "dev"
+    else:
+        doc_version = ".".join(sklearn.__version__.split(".")[:2])
```

Comment:
Can we reuse the code that generates the link for the documentation of the estimator itself ?

Reply:
I tried to use _get_doc_link(), it works nice when it's only a simple estimator. But when it's a pipeline for example, it doesn't. At least for now. So, I need to spend more time on it.

Reply:
Except that I still need to fix the tests.